jsonargparse_tests/test_core.py

```diff
@@ -801,7 +806,15 @@ def test_default_config_file_unreadable(parser, tmp_cwd):
 
     assert parser.get_default("op1") == "from yaml"
     default_config_file.chmod(0)
-    assert parser.get_default("op1") == "from default"
+    try:
+        # Check if the file is actually unreadable.
+        default_config_file.read_text()
+    except PermissionError:
+        assert parser.get_default("op1") == "from default"
+    else:
+        # Sometimes chmod 0 will not make the file unreadable
+        # (e.g. if running as root in a docker container).
+        pytest.skip("File was still readable even after chmod(0)")
```

Comment:
Please skip the entire test function using a decorator, instead of this try-except block.